### PR TITLE
VtGateV3 SEQUENCE tests in GO using new cluster

### PR DESCRIPTION
**Pull request by ajeetj**
VtGateV3 SEQUENCE tests in GO using the new cluster. Currently it's against the Cluster branch. 

I will close the old pr now.

### CODE REVIEW

File: go/test/endtoend/vtgate/seq_test.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	}
+
+	// Test insert with no auto-inc
+	exec(t, conn, "insert into sequence_test(id, val) values(6, 'f')")
+	qr = exec(t, conn, "select * from sequence_test")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[INT64(1) VARCHAR("a")] [INT64(2) VARCHAR("b")] [INT64(3) VARCHAR("c")] [INT64(4) VARCHAR("d")] [INT64(6) VARCHAR("f")]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	//Next INsert will fail as we have corrupted the sequence
```
Comment (deepthi) [inline go/test/endtoend/vtgate/seq_test.go]: INsert -> insert

File: go/test/endtoend/vtgate/seq_test.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	// Test next available seq id from cache
+	qr = exec(t, conn, "select next 1 values from sequence_test_seq")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[INT64(5)]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	//Test next_id from seq table. This will be alloted in case cache is blew up.
```
Comment (deepthi) [inline go/test/endtoend/vtgate/seq_test.go]: Can you fix this comment to say exactly what next_id is supposed to represent/ return?
Something like `As soon as we start using values from the cache, next_id is updated to the last value in the cache + 1` (or whatever is correct).

File: go/test/endtoend/vtgate/seq_test.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	"strings"
+	"testing"
+
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/test/endtoend/cluster"
+)
+
+var (
+	ClusterInstance *cluster.LocalProcessCluster
```
Comment (deepthi) [inline go/test/endtoend/vtgate/seq_test.go]: Change all these variables to start with lowercase. They should be used only in this file.

**Review by deepthi — APPROVED** [review]
A few minor changes requested. You can merge once they are done.

File: go/test/endtoend/vtgate/seq_test.go (hunk 3 above)
Comment (ajeetj) [reply go/test/endtoend/vtgate/seq_test.go]: Done

File: go/test/endtoend/vtgate/seq_test.go (hunk 1 above)
Comment (ajeetj) [reply go/test/endtoend/vtgate/seq_test.go]: Done

File: go/test/endtoend/vtgate/seq_test.go (hunk 2 above)
Comment (ajeetj) [reply go/test/endtoend/vtgate/seq_test.go]: Done

**Pull request merged**